```
    def test_chrome_webdriver_version(mocked_call):
        chrome = browser.Chrome(logger)
        webdriver_binary = '/usr/bin/chromedriver'
    
        # Working cases.
        mocked_call.return_value = 'ChromeDriver 84.0.4147.30'
>       assert chrome.webdriver_version(webdriver_binary) == '84.0.4147.30'
E       AssertionError: assert 'ChromeDriver 84.0.4147.30' == '84.0.4147.30'
E         - 84.0.4147.30
E         + ChromeDriver 84.0.4147.30
```

--- a/tools/wpt/browser.py
+++ b/tools/wpt/browser.py
@@ -728,6 +728,17 @@ class Chrome(Browser):
     def install_webdriver_by_version(self, version, dest=None):
         if dest is None:
             dest = os.pwd
+
+        # There may be an existing chromedriver binary from a previous install.
+        # To provide a clean install experience, remove the old binary - this
+        # avoids tricky issues like unzipping over a read-only file.
+        expected_binary_path = os.path.join(dest, 'chromedriver')
+        if os.path.isfile(expected_binary_path):
+            self.logger.info("Removing existing ChromeDriver binary: %s" %
+                expected_binary_path)
+            os.chmod(expected_binary_path, stat.S_IWUSR)
+            os.remove(expected_binary_path)
+
         url = self._latest_chromedriver_url(version) if version \
             else self._chromium_chromedriver_url(None)
         self.logger.info("Downloading ChromeDriver from %s" % url)
@@ -741,11 +752,11 @@ class Chrome(Browser):
             dest, 'chromedriver_%s' % self._chromedriver_platform_string())
         binary_path = find_executable("chromedriver", chromedriver_dir)
         if binary_path is not None:
-            shutil.copy(binary_path, dest)
+            shutil.move(binary_path, dest)
             rmtree(chromedriver_dir)
 
         binary_path = find_executable("chromedriver", dest)
-        assert binary_path is not None
+        assert binary_path == expected_binary_path
         return binary_path
 
     def install_webdriver(self, dest=None, channel=None, browser_binary=None):

--- a/tools/wpt/browser.py
+++ b/tools/wpt/browser.py
@@ -732,12 +732,12 @@ class Chrome(Browser):
         # There may be an existing chromedriver binary from a previous install.
         # To provide a clean install experience, remove the old binary - this
         # avoids tricky issues like unzipping over a read-only file.
-        expected_binary_path = os.path.join(dest, 'chromedriver')
-        if os.path.isfile(expected_binary_path):
+        existing_binary_path = find_executable("chromedriver", dest)
+        if existing_binary_path:
             self.logger.info("Removing existing ChromeDriver binary: %s" %
-                expected_binary_path)
-            os.chmod(expected_binary_path, stat.S_IWUSR)
-            os.remove(expected_binary_path)
+                existing_binary_path)
+            os.chmod(existing_binary_path, stat.S_IWUSR)
+            os.remove(existing_binary_path)
 
         url = self._latest_chromedriver_url(version) if version \
             else self._chromium_chromedriver_url(None)
@@ -756,7 +756,7 @@ class Chrome(Browser):
             rmtree(chromedriver_dir)
 
         binary_path = find_executable("chromedriver", dest)
-        assert binary_path == expected_binary_path
+        assert binary_path is not None
         return binary_path
 
     def install_webdriver(self, dest=None, channel=None, browser_binary=None):

--- a/tools/wpt/run.py
+++ b/tools/wpt/run.py
@@ -347,7 +347,6 @@ class Chrome(BrowserSetup):
                 install = self.prompt_install("chromedriver")
 
                 if install:
-                    logger.info("Downloading ChromeDriver")
                     webdriver_binary = self.browser.install_webdriver(
                         dest=self.venv.bin_path,
                         channel=browser_channel,

--- a/tools/wpt/tests/test_browser.py
+++ b/tools/wpt/tests/test_browser.py
@@ -59,6 +59,9 @@ def test_chrome_find_webdriver(mocked_find_executable):
     assert chrome.find_webdriver(browser_binary='/usr/bin/chrome') is None
 
 
+# On Windows, webdriver_version directly calls _get_fileversion, so there is no
+# logic to test there.
+@pytest.mark.skipif(sys.platform.startswith('win'), reason='just uses _get_fileversion on Windows')
 @mock.patch('tools.wpt.browser.call')
 def test_chrome_webdriver_version(mocked_call):
     chrome = browser.Chrome(logger)

